cmd/secrets-provider/main.go

```diff
@@ -51,63 +51,52 @@ func main() {
 		printErrorAndExit(messages.CSPFK009E)
 	}
 
-	// Configure exponential backoff
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 2 * time.Second
-	expBackoff.RandomizationFactor = 0.5
-	expBackoff.Multiplier = 2
-	expBackoff.MaxInterval = 15 * time.Second
-	expBackoff.MaxElapsedTime = 2 * time.Minute
-
+	// Configure constant backoff for application container
+	constantBackOff := backoff.NewConstantBackOff(time.Duration(secretsConfig.SyncIntervalTime) * time.Minute)
+	count := 0
 	err = backoff.Retry(func() error {
-		for {
-			log.Info(fmt.Sprintf(messages.CSPFK001I, authn.Config.Username))
-			authnResp, err := authn.Authenticate()
-			if err != nil {
-				return log.RecordedError(messages.CSPFK010E)
-			}
-
-			err = authn.ParseAuthenticationResponse(authnResp)
-			if err != nil {
-				return log.RecordedError(messages.CSPFK011E)
-			}
-
-			err = provideConjurSecrets(accessToken)
-			if err != nil {
-				return log.RecordedError(messages.CSPFK016E)
-			}
-
+		if count > secretsConfig.RetryCount {
```

Comment:
i have a suggestion that i think will make the code clearer. currently we return different errors in case they occurred and catch them after the backoff loop. Then, we clean up the token and print the errors.

However, not you add a new behaviour where you handle the error (retry count limit reached) inside the loop itself.

What you can do is return an error in this case and still catch all the errors outside of the loop.